neo/SmartContract/ApplicationEngine.cs

```diff
@@ -100,17 +112,104 @@ private bool CheckItemSize()
             }
         }
 
-        private bool CheckStackSize()
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="value">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        private bool CheckBigInteger(BigInteger value)
+        {
+            return value == null ? false :
+                value.ToByteArray().Length <= MaxSizeForBigInteger;
+        }
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary> 
+        private bool CheckBigIntegers(OpCode nextInstruction)
+        {
+            switch (nextInstruction)
+            {
+                case OpCode.INC:
+                    {
+                        BigInteger x = EvaluationStack.Peek().GetBigInteger();
+
+                        if (!CheckBigInteger(x) || !CheckBigInteger(x + 1))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.DEC:
+                    {
+                        BigInteger x = EvaluationStack.Peek().GetBigInteger();
+
+                        if (!CheckBigInteger(x) || !CheckBigInteger(x - 1))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.ADD:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 + x2))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.SUB:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 - x2))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.MUL:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 * x2))
```

Comment:
I think this check will consume as much resources as normal execution.

Reply:
Maybe the operation does not have to be verified?

Reply:
Or maybe in neo-vm could create virtual method like 
```protected virtual bool CheckMulOpCode(BigItenger a,BigInteger b);```

Reply:
Can we check the length of `x1` and `x2`, and then predict the length of result?